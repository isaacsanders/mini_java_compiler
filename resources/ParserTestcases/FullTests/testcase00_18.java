class MainClass1 {
	public static void main(String[] a) {
	}
}

class AnotherClass extends foo {
	int anint;
	boolean abool;
	bar abar;
}

class AndAnother {
	int wereallylikeints;
	public monkey mongoose(foo a, foo b) {
		return wereallylikeints;
	}
}
