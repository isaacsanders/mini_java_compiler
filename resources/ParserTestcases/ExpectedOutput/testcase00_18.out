StmtList ::= 
MainClassDecl ::= ...
ClassDeclList ::= 
OptExtends ::= EXTENDS ID
ClassVarDeclList ::= 
Type ::= INT
ClassVarDecl ::= Type ID SEMI
ClassVarDeclList ::= ClassVarDeclList ClassVarDecl
Type ::= BOOLEAN
ClassVarDecl ::= Type ID SEMI
ClassVarDeclList ::= ClassVarDeclList ClassVarDecl
Type ::= ID
ClassVarDecl ::= Type ID SEMI
ClassVarDeclList ::= ClassVarDeclList ClassVarDecl
MethodDeclList ::= 
ClassDecl ::= ...
ClassDeclList ::= ClassDeclList ClassDecl
OptExtends ::= 
ClassVarDeclList ::= 
Type ::= INT
ClassVarDecl ::= Type ID SEMI
ClassVarDeclList ::= ClassVarDeclList ClassVarDecl
MethodDeclList ::= 
Type ::= ID
Type ::= ID
Formal ::= Type ID
FormalList ::= Formal
Type ::= ID
Formal ::= Type ID
FormalList ::= FormalList COMMA Formal
OptFormalList ::= FormalList
StmtList ::= 
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
MethodDecl ::= ... 
MethodDeclList ::= MethodDeclList MethodDecl
ClassDecl ::= ...
ClassDeclList ::= ClassDeclList ClassDecl
Program ::= MainClassDecl ClassDeclList
